nautobot/utilities/tests/test_templatetags_netutils.py

```diff
@@ -0,0 +1,52 @@
+import inspect
+
+from django.template import Context, Engine
+from django.template.exceptions import TemplateSyntaxError
+from django.test import TestCase
+
+from netutils.utils import jinja2_convenience_function
+
+
+class NautobotTemplateTagsNetutilsTest(TestCase):
+    """Test the use of netutils functions as Django template filters."""
+
+    def test_netutils_filters_in_django(self):
+        """Verify that each function is at least available as a filter."""
+        engine = Engine.get_default()
+        context = Context({})
+        for filter_name, filter_func in jinja2_convenience_function().items():
+            with self.subTest(f'Testing filter "{filter_name}"'):
+                template_string = "{% load netutils %}{% " + filter_name
+                signature = inspect.signature(filter_func)
+                i = 1
+                for param_name, param in signature.parameters.items():
+                    template_string += f" {param_name}="
+                    if param.annotation == str:
+                        template_string += f'"{i}"'
+                    elif param.annotation == bool:
+                        template_string += "True"
+                    elif param.annotation in (int, float):
+                        template_string += str(i)
+                    elif param.annotation in (list, tuple):
+                        template_string += "[]"
+                    elif param.annotation == dict:
+                        template_string += "{}"
+                    else:
+                        template_string += "None"
+                    i += 1
+                template_string += " %}"
+                template = engine.from_string(template_string)
+                try:
+                    template.render(context)
+                except TemplateSyntaxError as exc:
+                    # Django doesn't see this as a valid template-tag, or similar - shouldn't happen
+                    self.fail(str(exc))
+                except (AttributeError, KeyError, TypeError, ValueError):
+                    # We didn't pass "valid" params to the function, but at least Django saw it as a template tag.
+                    pass
+                except ConnectionError:
+                    # Yes, some netutils functions such as tcp_ping actually make network calls when invoked. Eeek.
+                    pass
+                except Exception:
+                    # Catch-all - at least it wasn't a TemplateSyntaxError, so good enough for now.
+                    pass
```

Comment:
This is a step forward, but having all these expected error types and passing them through feels a little icky. Without boiling the ocean is there a way we could know which ones we expect to error or not and test those separately with `assertRaises`?

I don't want to block on this. More food for thought than anything.

Reply:
I agree, but it's a quick-and-dirty way to test all of the functions without needing to write separate tests per function (and update those when new netutils releases add more). I actually originally started with just `except TemplateError: fail; except Exception: pass` but tried to add more specific ones to see if I could use `inspect` to automatically provide "correct" inputs. That turned out to be unlikely, in particular because there's lots of functions that take `str` as input but actually expect/enforce that it's a string of an IP address or similar.

The alternate option would be to just test a few specific known functions (since we're using a big `for` loop anyway to load them, either they should all work or none will), which might be preferable?